artbotlib/pipeline_image_names.py

```diff
@@ -0,0 +1,422 @@
+    response = request_with_kerberos(url)
+
+    if response.status_code == 404:
+        raise CdnNotFound(cdn_name)
+
+    return response.json()
+
+
+def cdn_to_comet(cdn_name):
+    response = get_cdn_repo_details(cdn_name)
+
+    try:
+        return response['data']['attributes']['external_name']
+    except Exception:
+        raise DeliveryRepoNotFound(cdn_name)
+
+
+def get_cdn_repo_id(cdn_name):
+    response = get_cdn_repo_details(cdn_name)
+
+    try:
+        return response['data']['id']
+    except Exception:
+        raise CdnIdNotFound(cdn_name)
+
+
+def distgit_is_available(distgit_repo_name):
+    response = requests.head(f"https://pkgs.devel.redhat.com/cgit/containers/{distgit_repo_name}")
+    return response.status_code == 200
+
+
+def get_brew_id(brew_name):
+    """
+    Get the brew id for the given brew name.
+
+    :so: SlackOutput object for reporting results.
+    :brew_name: The name of the brew package
+    """
+    try:
+        koji_api = util.koji_client_session()
+    except Exception:
+        raise KojiClientError
+
+    try:
+        brew_id = koji_api.getPackageID(brew_name, strict=True)
+    except Exception:
+        raise BrewIdNotFound(brew_name)
+
+    return brew_id
+
+
+def get_variant_id(cdn_name, variant_name):
+    response = get_cdn_repo_details(cdn_name)
+
+    try:
+        for data in response['data']['relationships']['variants']:
+            if data['name'] == variant_name:
+                return data['id']
+    except Exception:
+        raise VariantIdNotFound(cdn_name, variant_name)
+
+
+def get_product_id(variant_id):
+    url = f"https://errata.devel.redhat.com/api/v1/variants/{variant_id}"
+    response = request_with_kerberos(url)
+
+    try:
+        return response.json()['data']['attributes']['relationships']['product_version']['id']
+    except Exception:
+        raise ProductIdNotFound(variant_id)
+
+
+def get_delivery_repo_id(name):
+    url = f"https://pyxis.engineering.redhat.com/v1/repositories?filter=repository=={name}"
+    response = request_with_kerberos(url)
+
+    if response.status_code == 404:
+        raise DeliveryRepoUrlNotFound
+
+    try:
+        repo_id = response.json()['data'][0]['_id']
+    except Exception:
+        raise DeliveryRepoIDNotFound(name)
+
+    return repo_id
+
+
+def github_distgit_mappings(version):
+    output = util.cmd_gather("doozer -g openshift-" + version + " images:print --short '{name}: {upstream_public}'")
+    data = output[1].split("\n")
+    dict_data = {}
+    for line in data:
+        array = line.split(": ")
+        if array != ['']:
+            dict_data[array[1]] = array[0]
+    return dict_data
+
+
+def distgit_github_mappings(version):
+    output = util.cmd_gather("doozer -g openshift-" + version + " images:print --short '{name}: {upstream_public}'")
+    data = output[1].split("\n")
+    dict_data = {}
+    for line in data:
+        array = line.split(": ")
+        if array != ['']:
```

Comment:
seems overly specific. this shouldn't happen with this output and `splitlines()`, but if it did, i'd probably want to ignore more:
```suggestion
        if len(array) == 2:
```